### PR TITLE
Feature/203468 handle customer cancel payment

### PR DESCRIPTION
**Pull request by C4rter**
Handle new payment aborted by customer status.
Also fix missing checkout form data in case of payment error.

### CODE REVIEW

File: checkout/interfaces/controller/checkoutcontroller.go [hunk 1]
```diff
@@ -248,6 +248,9 @@ func (cc *CheckoutController) PlaceOrderAction(ctx context.Context, r *web.Reque
 	if err != nil {
 		cc.logger.WithContext(ctx).WithField("subcategory", "checkoutError").WithField("errorMsg", err.Error()).Error(fmt.Sprintf("place order failed: cart id: %v / total-amount: %v", decoratedCart.Cart.EntityID, decoratedCart.Cart.GrandTotal()))
 		if paymentError, ok := err.(*paymentDomain.Error); ok {
+			if paymentError.ErrorCode == "payment-aborted-by-customer" {
+				return cc.showCheckoutFormAndHandleSubmit(ctx, r, "checkout/checkout")
+			}
```
Comment (jeinfeldt) [inline checkout/interfaces/controller/checkoutcontroller.go]: Could you move the "payment-aborted-by-customer" to a constant?

Comment (C4rter) [reply checkout/interfaces/controller/checkoutcontroller.go]: Created a constant for this now.

**Comment by C4rter** [comment]
> Please verify all travis checks pass

All passed now

File: checkout/interfaces/controller/checkoutcontroller.go (hunk 1 above)
Comment (C4rter) [reply checkout/interfaces/controller/checkoutcontroller.go]: Moved constant to payment domain.

**Pull request merged**